File: host/python/src/one_sdk/handle_map.py
from typing import Generic, MutableMapping, Optional, TypeVar

T = TypeVar("T")
class HandleMap(Generic[T]):
	def __init__(self):
		self._index: int = 1
		self.data: MutableMapping[int, T] = {}
	
	def insert(self, value: T) -> int:
		handle = self._index
		self._index += 1
		self.data[handle] = value

		return handle
	
	def get(self, handle: int) -> Optional[T]:
		return self.data.get(handle, None)
	
	def remove(self, handle: int) -> Optional[T]:
		return self.data.pop(handle, None)

File: packages/javascript_common/src/lib/handle_map.ts
export class HandleMap<T> {
  private index: number = 1;
  private data: Record<number, T> = {};

  insert(value: T): number {
    const handle = this.index;
    this.index += 1;
    this.data[handle] = value;

    return handle;
  }

  get(handle: number): T | undefined {
    return this.data[handle];
  }

  remove(handle: number): T | undefined {
    const value = this.get(handle);
    if (value !== undefined) {
      delete this.data[handle];
    }

    return value;
  }
}
